muxers/yamux/src/lib.rs

```diff
@@ -252,62 +283,69 @@ impl WindowUpdateMode {
     /// > **Note**: With this strategy, there is usually no point in the
     /// > receive buffer being larger than the window size.
     pub fn on_read() -> Self {
-        WindowUpdateMode(yamux::WindowUpdateMode::OnRead)
+        WindowUpdateMode(yamux012::WindowUpdateMode::OnRead)
     }
 }
 
 impl Config {
     /// Creates a new `YamuxConfig` in client mode, regardless of whether
     /// it will be used for an inbound or outbound upgrade.
+    #[deprecated(note = "Will be removed with the next breaking release.")]
     pub fn client() -> Self {
-        Self {
-            mode: Some(yamux::Mode::Client),
+        Self(Either::Left(Config012 {
+            mode: Some(yamux012::Mode::Client),
             ..Default::default()
-        }
+        }))
     }
 
     /// Creates a new `YamuxConfig` in server mode, regardless of whether
     /// it will be used for an inbound or outbound upgrade.
+    #[deprecated(note = "Will be removed with the next breaking release.")]
     pub fn server() -> Self {
-        Self {
-            mode: Some(yamux::Mode::Server),
+        Self(Either::Left(Config012 {
+            mode: Some(yamux012::Mode::Server),
             ..Default::default()
-        }
+        }))
     }
 
-    /// Sets the size (in bytes) of the receive window per substream.
-    pub fn set_receive_window_size(&mut self, num_bytes: u32) -> &mut Self {
-        self.inner.set_receive_window(num_bytes);
+    fn set(&mut self, f: impl FnOnce(&mut yamux012::Config) -> &mut yamux012::Config) -> &mut Self {
```

Comment:
Preference for functions to be sorted after the visibility and overall call-order with private functions below all public ones.